libs/community/langchain_community/document_loaders/sqlalchemy.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict, List, Optional, Union
+
+import sqlalchemy as sa
+
+from langchain_community.docstore.document import Document
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class SQLAlchemyLoader(BaseLoader):
+    """
+    Load documents by querying database tables supported by SQLAlchemy.
+    Each document represents one row of the result.
+    """
+
+    def __init__(
+        self,
+        query: Union[str, sa.Select],
+        url: str,
+        page_content_columns: Optional[List[str]] = None,
+        metadata_columns: Optional[List[str]] = None,
+        source_columns: Optional[List[str]] = None,
+        include_rownum_into_metadata: bool = False,
+        include_query_into_metadata: bool = False,
```

Comment:
I think it would be more powerful to be able to pass a function to convert a row to str for the page_content and a function to convert a row to Dict[str, Any] for the metadata.

Reply:
That is an excellent suggestion, thank you Christophe. I will try to reshape this spot into the direction you've outlined.

Reply:
Dear Christophe. I've amended the patch as you suggested, and just pushed the outcome. CI does not complain about it, so I will be happy to receive another review on it.